### ISSUE TITLE
[Feature Request]: option to hide ComposedModal header close icon

### ISSUE THREAD

**Issue by makafsal** [carbon-design-system/carbon#18262]
### The problem

Currently, we hide the `ComposedModal` close icon using styles, but hiding the close button with styles can sometimes cause issues. Providing a property or a way to pass a custom close icon would help eliminate these issues.

Also, placing the `children` in the `ModalHeader` component above the close button causes a tab order issue (wrong hierarchical order). Please see the issue: [6523](https://github.com/carbon-design-system/ibm-products/issues/6523)

![Image](https://github.com/user-attachments/assets/ee1fcf7a-499b-4548-a89d-0dfa6451f365)

### The solution

**Approach 1**
1. Provide a property `hasCloseIcon` or `hideCloseIcon` to hide the close button
2. Move the `children` to below the close icon, and move the icon button to the top (as the first element in the header)

**Approach 2**
1. Remove the current close button and allow to pass a custom close button with `children`

### Application/PAL

Carbon for IBM Products

### Business priority

Low Priority = release date is not dependent on fix or not upcoming

### Available extra resources

https://github.com/carbon-design-system/ibm-products/issues/6523

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

**Comment by sstrubberg** [comment]
We're not going to add a prop and we can't move children because that would be breaking. But we could move the children and put it behind a flag.

From a design perspective, do we allow the button to be hidden? It's in the anatomy of the modal.

**Comment by tay1orjones** [comment]
Yeah, right now the guidance specifies the [anatomy](https://carbondesignsystem.com/components/modal/usage/#anatomy) of the modal includes a non-optional close button. If you want to hide this button you'll need to reimplement `ModalHeader` on your own to handle that behavior.

I do think this could be more composable though in the future. Maybe we could explore that behind a feature flag. Breaking out the close IconButton into it's own component and when the flag is on it wouldn't render it. This would give you greater control:

```js
<FeatureFlag ... />
  <ModalHeader>
    {your content without the close button}
    <ModalHeaderCloseButton />
  </ModalHeader>
</FeatureFlag>
```
